discover/portal/articles/220-polls.markdown

```diff
@@ -62,53 +55,52 @@ remove an expired poll from a page, remove the Poll Display portlet or
-users to vote on a certain poll question, while others can only view it. For
-further information about permissions, please see chapters 15 and 16. 
+*Permissions* are set on individual polls. Use permissions, for example, to
+allow some privileged users to vote on a certain poll question, while others can
+only view it. 
 
-As you can see, creating a poll is fairly straightforward. Next, let's complete
-the two-step process and put your poll on a page.
+Creating a poll is fairly straightforward. Next, complete the two-step
+process and put your poll on a page.
 
 ## Adding a Poll to a Page [](id=adding-a-poll-to-a-page)
 
 Now that you have created your poll question, it's time to present it to your
 users. Follow these steps.
 
-1.  Go to the page where you want to display poll results and add a Polls
-    Display widget from *Add* &rarr; *Widgets* &rarr; *Content Management*.
+1.  Go to a page and add a Polls Display widget from *Add* &rarr; *Widgets*
+    &rarr; *Content Management*.
 
 2.  Click *Please configure this portlet to make it visible to all users.*
 
-3.  In the dialog box that appears, select the title of the poll you want to
-    display. Click *Save*.
+3.  In the dialog box that appears, select the title of the poll to display.
 
-Once the poll question has been successfully placed on the page,you can perform
+4.  Click *Save*.
+
+Once the poll question has been successfully placed on the page, perform
 other tasks using the icons at the bottom of the portlet.
 
-![Figure 2: These buttons provide shortcuts to the widgets configuration, as well
-as to some of the Polls Portlet's functionality.](../images/poll-buttons.png)
+![Figure 2: These buttons provide shortcuts to the widget's configuration, as well
+as to some of the Polls Application's functionality.](../images/poll-buttons.png)
 
 **Edit Question:** Displays a similar dialog box to the one used to create the
 poll. 
 
 **Select Poll:** Displays the same dialog box as Configuration, allowing you to
-choose different questions from the drop-down menu. 
+choose different polls from the drop-down menu. 
 
 **Add:** Allows you to create a new poll. 
 
-## Viewing the Poll Results [](id=viewing-the-poll-results)
+## Viewing Poll Results [](id=viewing-the-poll-results)
 
 All the polls you create are listed in the Polls portlet in *Site
 Administration* &rarr; *Content* &rarr; *Polls*. When users vote in the poll,
 the data is collected here. Click on a poll see a breakdown of the results.
 
 ![Figure 3: Selecting a poll in the Polls portlet puts the data at your fingertips.](../images/polls-results.png)
 
-If you click on one of the listed *Charts*, the portlet will generate an
+If you click on one of the listed *Charts*, the portlet generates an
 appropriate visualization of the data.
 
-
 Below this is an item called *Charts*. This option shows the poll results
 represented in various graphs. The graphs are *Area*, *Horizontal Bar*, *Line*,
 *Pie*, and *Vertical Bar*.
```

Comment:
In the last paragraph, after the following screenshot, it's mentioned that Guest users are represented by numbers, however during my local tests no number or name was assigned to the votes of Guest users. Have you had the same issue?

Reply:
@natocesarrego  Yes, it's blank for me too, but then I see logic in `question_results.jspf` that makes me wonder why there's no numeric ID displayed:
```		List resultRows = searchContainer.getResultRows();

		for (int i = 0; i < results.size(); i++) {
			PollsVote vote = (PollsVote)results.get(i);

			ResultRow row = new ResultRow(vote, vote.getVoteId(), i);

			// User

			User user2 = null;

			try {
				user2 = UserLocalServiceUtil.getUserById(vote.getUserId());
			}
			catch (Exception e) {
			}

			if (user2 != null) {
				row.addText(HtmlUtil.escape(user2.getFullName()), user2.isActive() ? user2.getDisplayURL(themeDisplay) : null);
			}
			else {
				row.addText(String.valueOf(vote.getUserId()));
			}

```

In my database each anonymous vote has a user id of 20103, and a username of Anonymous. It might make sense to display Anonymous for each of these votes.

Reply:
I've just checked my database and I've seen the same user. This is the default user which comes with Liferay by default and once he doesn't have a name, an empty name is displayed. 🙂